Sources/MapboxDirections/Incident.swift

```diff
@@ -21,46 +20,174 @@ public struct Incident: Codable, Equatable {
         case geometryIndexStart = "geometry_index_start"
         case geometryIndexEnd = "geometry_index_end"
     }
-
-    var identifier: String
-    var type: String
-    var description: String
-    var creationTime: String
-    var startTime: String
-    var endTime: String
-    var impact: String?
-    var subtype: String?
-    var subtypeDescription: String?
-    var alertCodes: [Int]
-    var lanesBlocked: [Int]
-    var geometryIndexStart: Int
-    var geometryIndexEnd: Int
+    
+    /// Defines known types of incidents.
+    ///
+    /// Each incident may or may not have specific set of data, depending on it's `kind`
+    public enum Kind: String {
+        case Accident = "accident"
+        case Congestion = "congestion"
+        case Construction = "construction"
+        case DisabledVehicle = "disabled_vehicle"
+        case LaneRestriction = "lane_restriction"
+        case MassTransit = "mass_transit"
+        case Miscellaneous = "miscellaneous"
+        case OtherNews = "other_news"
+        case PlannedEvent = "planned_event"
+        case RoadClosure = "road_closure"
+        case RoadHazard = "road_hazard"
+        case Weather = "weather"
+    }
+    
+    /// Defines a lane affected by the `Incident`
+    ///
+    /// Each `Incident` may have arbitrary of affected lanes
+    public enum BlockedLane: String, Codable {
+        case Left = "LEFT"
+        case LeftCenter = "LEFT CENTER"
+        case LeftTurnLane = "LEFT TURN LANE"
+        case Center = "CENTER"
+        case Right = "RIGHT"
+        case RightCenter = "RIGHT CENTER"
+        case RightTurnLane = "RIGHT TURN LANE"
+        case Hov = "HOV"
+        case Side = "SIDE"
+        case Shoulder = "SHOULDER"
+        case Median = "MEDIAN"
+        case Lane1 = "1"
+        case Lane2 = "2"
+        case Lane3 = "3"
+        case Lane4 = "4"
+        case Lane5 = "5"
+        case Lane6 = "6"
+        case Lane7 = "7"
+        case Lane8 = "8"
+        case Lane9 = "9"
+        case Lane10 = "10"
```

Comment:
For the documentation comments for these cases, let’s say e.g. “Second lane from the left”, assuming that’s what `lane2` means even in a region that drives on the left.

Reply:
Doesn't "Lane1" means the leftmost lane for Left Driving Side regions and the rightmost lane for Right Driving Side regions?

Reply:
I was assuming the numbering follows the California standard, but other regions may number the lanes differently, so maybe we need to be ambiguous for now.

![](https://www.dmv.ca.gov/imageserver/dmv/images/dlhdbk/71_240.gif)

Reply:
The National Traffic Information Management Coalition, which is affiliated with AASHTO in the U.S., has been [recommending](https://www.michigan.gov/documents/mdot/13-NTIMC_Lane_Designation_Terminology_517666_7.pdf) the California lane numbering scheme for use nationwide.